packages/nimble-components/src/table-column/mixins/custom-sort-order.ts

```diff
@@ -0,0 +1,96 @@
+import { Notifier, Observable, attr } from '@microsoft/fast-element';
+import type { TableColumn } from '../base';
+import { TableColumnSortOperation } from '../base/types';
+import type { ColumnValidator } from '../base/models/column-validator';
+
+// Pick just the relevant properties the mixin depends on (typescript complains if the mixin declares private / protected base exports)
+type CustomSortOrderTableColumn<
+    TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>
+> = Pick<TableColumn<unknown, TColumnValidator>, 'columnInternals'>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type CustomSortOrderTableColumnConstructor<TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>> = abstract new (...args: any[]) => CustomSortOrderTableColumn<TColumnValidator>;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinCustomSortOrderColumnAPI<
+    TBase extends CustomSortOrderTableColumnConstructor<TColumnValidator>,
+    TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow sorting
+     * by a field other than the default for that column.
+     */
+    abstract class CustomSortOrderColumn extends base {
+        public sortByFieldName?: string;
+
+        /** @internal */
+        public notifier?: Notifier;
```

Comment:
Need more uniqueness to some of these properties. They are defined on the prototype chain so this would be exposed as textColumnRef.notifier and and textColumnRef.handleChange. Seems like it would be pretty easy to mixin on a class and conflict / override properties with these names.

We may want a separate handler object instead of adding handleChange to the class we are mixing in on.